ui/apps/platform/cypress/integration/audit/listeningEndpoints/ListeningEndpoints.helpers.js

```diff
@@ -0,0 +1,10 @@
+import { visit } from '../../../helpers/visit';
+
+const basePath = '/main/audit/listening-endpoints/';
+
+export function visitListeningEndpoints() {
+    visit(basePath);
+
+    cy.get('h1:contains("Listening endpoints")');
+    cy.location('pathname').should('eq', basePath);
```

Comment:
I'm starting to lean into not having the `visit...X` functions make any assertions, for a couple of reasons.

1. We reuse the visit functions on multiple tests, and it's just wasting cycles--albeit a small amount--to test the same thing over and over.
2. It violates the principle that a function should have only one "function".

Reply:
Good points. Wasted cycles in tests has been on my mind lately too. My assumption in the past has been to not worry about selector or assertion performance because the time spent in those calls should be minimal compared to the overhead of having to load the app and wait for a group of network requests, but I haven't benchmarked it to be sure. I would be curious to see where the time was spent in an typical cypress test suite and see if there are any easy areas of improvement.

An argument for keeping the assertions here is that a failure could indicate a change in permissions, routing, etc. that otherwise might not be as evident when errors occur deeper in the test. For me though, when that type of thing happens I usually go directly to the video output, so it's probably only a minor improvement.

Reply:
With respect, I disagree.

After investigating more than one hundred failures in tests that worked for many years in favorable situations before beginning to fail in unfavorable situations:
1. On one side of the coin, let’s follow some robust patterns:
    * For example, giving cypress opportunities to wait on obvious generic assertions if central slows down instead of failing on test-specific assertions.
    * Especially if we prefer not to wait on responses, although that deserves discussion among the team.
    * Furthermore, although there are pro and con with judgment calls for selectors versus helpers, encapsulating some selectors in semantic helper functions reduces the amount of hunting for definitions when investigating or maintaining tests. For example, when the title changes in left navigation or landing page.
3. On other side of the coin, why make a new set of decisions for every container? 80-20 principle:
    * Reuse reliable patterns for 20% of decisions that apply to 80% of situations.
    * Invest mental creativity and energy in container-specific test design and implementation.

Reply:
I don't follow the statement ` indicate a change in permissions, routing, etc.`.

We can still make the assertion that the header is there, at the beginning of the first test. Likewise, if permissions change or block a view, we should mock the limited permissions in a different test to verify

Reply:
I was thinking along the lines of Mark's comment `giving cypress opportunities to wait on obvious generic assertions if central slows down instead of failing on test-specific assertions.`, but in the cases I listed above having the header assertion at the beginning of any test in the suite would catch the problem in the same way.

Agreed on the path assertion - that is out of place in this direct navigation helper and I'll remove it in the next PR.